File: 수학(구현)/백준_2847_게임을 만든 동준이/게임을 만든 동준이.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args)throws IOException {
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        int[] arr = new int[100];
        for(int i = 0; i < n; i++) arr[i] = s.nextInt();

        int sum = 0;
        for(int i = n-1; i>0; i--){
            if(arr[i-1] >= arr[i]){
                sum += (arr[i-1] - (arr[i] - 1));
                arr[i-1] = arr[i] - 1;
            }
        }
        System.out.println(sum);
    }
}

File: 수학(구현)/백준_2847_게임을 만든 동준이/게임을 만든 동준이.kt
import java.util.*

fun main(args: Array<String>) {
    val s = Scanner(System.`in`)
    val n = s.nextInt()
    var arr = IntArray(100, {0})

    for(i in 0 until n) arr[i] = s.nextInt()

    var sum = 0
    for(i in n-1 downTo 1){
        if(arr[i-1] >= arr[i]){
            sum += (arr[i-1] - (arr[i] - 1))
            arr[i-1] = arr[i] - 1
        }
    }
    print(sum)

}